cmd/util.go

```diff
@@ -32,280 +32,106 @@ func isInNamespace(meta metav1.ObjectMeta, namespace string) (valid bool) {
 	return namespace == apiv1.NamespaceAll || namespace == meta.Namespace
 }
 
-func convertDeploymentToDeploymentList(deployment Deployment) (deploymentList *DeploymentList) {
-	if isInRootConfigNamespace(deployment.ObjectMeta) {
-		deploymentList = &DeploymentList{Items: []Deployment{deployment}}
-	} else {
-		deploymentList = &DeploymentList{Items: []Deployment{}}
+func getContainers(resource k8sRuntime.Object) (container []Container) {
+	switch kubeType := resource.(type) {
+	case *StatefulSet:
+		container = kubeType.Spec.Template.Spec.Containers
+	case *DaemonSet:
+		container = kubeType.Spec.Template.Spec.Containers
+	case *ReplicationController:
+		container = kubeType.Spec.Template.Spec.Containers
+	case *Deployment:
+		container = kubeType.Spec.Template.Spec.Containers
+	case *Pod:
+		container = kubeType.Spec.Containers
+	}
+	return container
+}
+
+func newResultFromResource(resource k8sRuntime.Object) (result Result) {
+	switch kubeType := resource.(type) {
+	case *StatefulSet:
+		result.Name = kubeType.Name
+		result.Namespace = kubeType.Namespace
+		result.KubeType = kubeType.String()
```

Comment:
This results in the whole object being printed, which is rather messy.
The best alternative I could come up with is `result.KubeType = strings.Split(reflect.TypeOf(kubeType).String(), ".")[1]`

Reply:
Good catch, I will just go with the old solution and put the strings back in, since we need a case for every type there is no need for the `reflect` here.